### PR TITLE
[CALCITE-6674] Make RelDecorrelator rules configurable

### PR DESCRIPTION
**Pull request by rubenada**
See https://issues.apache.org/jira/browse/CALCITE-6674

### CODE REVIEW

**Review by mihaibudiu — APPROVED** [review]
Frankly I never understood why the decorrelator is special and different from the other optimization passes. You could claim it's a pre-packaged set of optimizations.

File: core/src/test/java/org/apache/calcite/sql2rel/RelDecorrelatorTest.java [hunk 1]
```diff
@@ -175,4 +180,86 @@ public static Frameworks.ConfigBuilder config() {
         + "                LogicalTableScan(table=[[scott, EMP]])\n";
     assertThat(after, hasTree(planAfter));
   }
+
+  @Test void testDecorrelatorCustomizeRules() {
```
Comment (caicancai) [inline core/src/test/java/org/apache/calcite/sql2rel/RelDecorrelatorTest.java:184]: Add jira link

**Review by caicancai — APPROVED** [review]

File: core/src/test/java/org/apache/calcite/sql2rel/RelDecorrelatorTest.java (hunk 1 above)
Comment (rubenada) [reply core/src/test/java/org/apache/calcite/sql2rel/RelDecorrelatorTest.java:189]: Done

**Comment by mihaibudiu** [comment]
Removed the "merge" label until the discussion in JIRA is settled.

**Comment by rubenada** [comment]
Replaced by https://github.com/apache/calcite/pull/4042

**Pull request closed without merging**